collectors/python.d.plugin/freeswitch/freeswitch.chart.py

```diff
@@ -0,0 +1,161 @@
+    'profiles'
+]
+
+
+CHARTS = {
+    'calls': {
+        'options': [None, 'Active Calls', 'calls', 'calls', 'freeswitch.calls', 'line'],
+        'lines': [
+            ['calls','calls','absolute']
+        ]
+    },
+    'registrations': {
+        'options': [None, 'Registrations', 'registrations', 'registrations', 'freeswitch.registrations', 'line'],
+        'lines': [
+            ['registrations','registrations','absolute']
+        ]
+    },
+    'gateways': {
+        'options': [None, 'Gateways', 'gateways', 'gateways', 'freeswitch.gateways', 'line'],
+        'lines': [
+            ['UNREGED','UNREGED','absolute'],
+            ['TRYING','TRYING','absolute'],
+            ['REGISTER','REGISTER','absolute'],
+            ['REGED','REGED','absolute'],
+            ['UNREGISTER','UNREGISTER','absolute'],
+            ['FAILED','FAILED','absolute'],
+            ['FAIL_WAIT','FAIL_WAIT','absolute'],
+            ['EXPIRED','EXPIRED','absolute'],
+            ['NOREG','NOREG','absolute'],
+            ['TIMEOUT','TIMEOUT','absolute'],
+        ]
+    },
+    'profiles': {
+        'options': [None, 'Active Profiles', 'profiles', 'profiles', 'freeswitch.profiles', 'line'],
+        'lines': [
+            ['RUNNING','RUNNING','absolute'],
+            ['DOWN','DOWN','absolute'],
+        ]
+    }
+}
+
+
+class Service(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+
+        self.order = ORDER
+        self.definitions = CHARTS
+
+        self.host = configuration.get('host', DEFAULT_HOST)
+        self.port = configuration.get('port', DEFAULT_PORT)
+        self.password = configuration.get('password', DEFAULT_PASSWORD)
+
+
+    def check(self):
+        if not HAS_ESL:
+            self.error("'ESL' package is needed to use freeswitch module, you can install with with pip install python-ESL")
+            return False
+
+        try:
+            self.fs = ESL.ESLconnection(self.host, self.port, self.password)
+            if not self.fs.connected():
+                raise ValueError('Not connected')
+        except ValueError:
+            self.error("Cannot connect to FreeSwitch with provided data, check that the service is running")
+            return False
+        return True
+
+    def _get_data(self):
+        if not self.fs.connected():
+            self.fs = ESL.ESLconnection(self.host, self.port, self.password)
+
+        try:
+            data = dict()
+
+            data['calls']=self.get_calls()
+            data['registrations']=self.get_registrations()
+
+            gateways = self.get_gateways()
+            profiles = self.get_profiles()
+
+            for gw_state in gateways:
+                data[gw_state]=int(gateways[gw_state])
+
+            for prof_state in profiles:
+                data[prof_state]=int(profiles[prof_state])
+
+            return data or None
+        except (ValueError, AttributeError):
```

Comment:
Where ValueError, AttributeError can happen?

Reply:
The conversions to int could throw a `ValueError`, and any of the dict accesses could throw an `AttributeError`.

However, this really should be better scoped, and it should be logging that an error happened if we're going to the trouble of catching them.